pkg/deployment/deployment.go

```diff
@@ -334,15 +334,23 @@ func (d *Deployer) addCertMounts(
 func (d *Deployer) addServiceExtraMounts(
 	service dataplanev1.OpenStackDataPlaneService,
 ) (*dataplanev1.AnsibleEESpec, error) {
+	log := d.Helper.GetLogger()
 	client := d.Helper.GetClient()
 	baseMountPath := path.Join(ConfigPaths, service.Name)
 
-	for _, cmName := range service.Spec.ConfigMaps {
+	for _, serviceCM := range service.Spec.ConfigMaps {
 
 		volMounts := storage.VolMounts{}
 		cm := &corev1.ConfigMap{}
-		err := client.Get(d.Ctx, types.NamespacedName{Name: cmName, Namespace: service.Namespace}, cm)
-		if err != nil {
+		err := client.Get(d.Ctx, types.NamespacedName{Name: serviceCM.Name, Namespace: service.Namespace}, cm)
+		if err != nil && k8s_errors.IsNotFound(err) {
+			if serviceCM.Required {
+				return d.AeeSpec, err
```

Comment:
OK this looks good. If a required CM is missing when we start the deployment the deployment will fail. That still allows creating the service first with a CM name, then later generating the CM itself, and finally starting the deployment. This is needed as the nova service might be created earlier than the control plane generating the cell config secret for it.